crates/burn-tensor/src/tensor/api/numeric.rs

```diff
@@ -2030,6 +2030,106 @@ where
         // Assign the original tensor data to the appropriate slice of the padded tensor
         padded_tensor.slice_assign(ranges, self)
     }
+    /// Create a one hot tensor.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::Tensor;
+    ///
+    /// fn example<B: Backend>(){
+    ///     let device = Default::default();
+    ///     let indices: Tensor<B, 1> = Tensor::from_floats([0.0, 1.0, 2.0, 3.0], &device);
+    ///     let one_hot: Tensor<B, 4> = indices.one_hot(4);
```

Comment:
Shouldn't the output be a 2D tensor?

```suggestion
    ///     let one_hot: Tensor<B, 2> = indices.one_hot(4);
```

Reply:
For this reason, I think we need to add this condition to the `TensorCheck`:

> Tensor of rank one greater than input tensor 'indices', i.e. rank(output) = rank(indices) + 1

In other words/code:
```rust
if D + 1 != D2
```

Reply:
Thank you for your comment, I modified and added check function accordingly.